README.md

```diff
@@ -62,59 +62,115 @@ This will put the plugin binary in the `bin` and `$GOPATH/bin` folders. `dev`
 mode will only generate the binary for your platform and is faster:
 
 ```sh
-$ make
 $ make dev
 ```
 
 Put the plugin binary into a location of your choice. This directory
-will be specified as the [`plugin_directory`](https://www.vaultproject.io/docs/configuration/index.html#plugin_directory)
-in the Vault config used to start the server.
+will be specified as the [`plugin_directory`](https://www.vaultproject.io/docs/configuration#plugin_directory)
+in the Vault config used to start the server. It may also be specified
+via [`-dev-plugin-dir`](https://developer.hashicorp.com/vault/docs/commands/server#dev-plugin-dir)
+if running Vault in dev mode.
 
-```json
-...
+```hcl
+# config.hcl
 plugin_directory = "path/to/plugin/directory"
 ...
 ```
 
+### Register Plugin
+
 Start a Vault server with this config file:
+
 ```sh
-$ vault server -config=path/to/config.json ...
+$ vault server -dev -config=path/to/config.hcl ...
 ...
 ```
 
-Once the server is started, register the plugin in the Vault server's [plugin catalog](https://www.vaultproject.io/docs/internals/plugins.html#plugin-catalog):
+Or start a Vault server in dev mode:
 
 ```sh
-$ vault write sys/plugins/catalog/azure \
-        sha_256=<expected SHA256 Hex value of the plugin binary> \
-        command="vault-plugin-auth-azure"
-...
-Success! Data written to: sys/plugins/catalog/azure
+$ vault server -dev -dev-root-token-id=root -dev-plugin-dir="path/to/plugin/directory"
 ```
 
-Note you should generate a new sha256 checksum if you have made changes
-to the plugin. Example using openssl:
+Once the server is started, register the plugin in the Vault server's [plugin catalog](https://www.vaultproject.io/docs/plugins/plugin-architecture#plugin-catalog):
 
 ```sh
-openssl dgst -sha256 $GOPATH/vault-plugin-auth-azure
+$ SHA256=$(openssl dgst -sha256 bin/vault-plugin-auth-azure | cut -d ' ' -f2)
+$ vault plugin register \
+        -sha256=$SHA256 \
+        -command="vault-plugin-auth-azure" \
+        auth azure-plugin
 ...
-SHA256(.../go/bin/vault-plugin-auth-azure)= 896c13c0f5305daed381952a128322e02bc28a57d0c862a78cbc2ea66e8c6fa1
+Success! Data written to: sys/plugins/catalog/azure-plugin
 ```
 
-Enable the auth plugin backend using the Azure auth plugin:
+Finally, enable the auth method to use this plugin:
 
 ```sh
-$ vault auth enable -plugin-name='azure' plugin
+$ vault auth enable azure-plugin
 ...
 
-Successfully enabled 'plugin' at 'azure'!
+Successfully enabled 'plugin' at 'azure-plugin'!
+```
+
+### Azure Environment Setup
+
+A Terraform [configuration](bootstrap/terraform) is included in this repository that
+automates provisioning of Azure resources necessary to configure and authenticate
+using the auth method.
```

Comment:
nit: Might be good to include a list of resources that this config will create? Something like a bulleted list similar to the VLE: https://github.com/hashicorp/vault-lambda-extension/tree/main/quick-start#vault-lambda-extension-quick-start

Reply:
Good idea! I'll include something similar to that 👍

Reply:
Added in [52970b2](https://github.com/hashicorp/vault-plugin-auth-azure/pull/65/commits/52970b2117fb589bb36c0959d27ad9b03bb3855a). Thanks!